src/inference/src/dev/isync_infer_request.cpp

```diff
@@ -263,10 +263,10 @@ void ov::ISyncInferRequest::allocate_tensor(const ov::Output<const ov::Node>& po
 void ov::ISyncInferRequest::check_tensors() const {
     const auto& inputs = m_compiled_model->inputs();
     for (size_t i = 0; i < inputs.size(); i++) {
-        check_tensor(inputs[i], m_tensors.at(inputs[i].get_tensor_ptr()));
+        check_tensor(inputs[i], get_ref_tensor(inputs[i]));
```

Comment:
get_ref_tensor() can avoid tensor mismatch issue, but `m_tensors.at(inputs[i].get_tensor_ptr())` itself should not have any problem?

Reply:
We can see that the compiled_mode are different:
1. original model:
![image](https://github.com/riverlijunjie/openvino-1/assets/31196718/ab5fd4e8-214e-45cc-b971-33f18e42d564)

2. IF not create input/output port, the final compile model:
![image](https://github.com/riverlijunjie/openvino-1/assets/31196718/003ec31d-2d6c-434e-8609-55096380ba0e)

3. If create input/output port, the final compile model, do in this PR, the final compile model will keep no change after tranformation

![image](https://github.com/riverlijunjie/openvino-1/assets/31196718/36eb05c3-a6a4-4512-aa7d-fbb3ed57aeae)

It seems that create input/output port will effect on the transformation, it is surprising.

Reply:
did you remove the lines of set_names?